src/View/Helper/BreadcrumbsHelper.php

```diff
@@ -0,0 +1,268 @@
+     * and inserts the supplied callable before it.
+     *
+     * @param string $matchingTitle The title of the crumb you want to insert this one after
+     * @param string $title Title of the crumb
+     * @param string|array|null $link Link of the crumb. Either a string, an array of route params to pass to
+     * Url::build() or null / empty if the crumb does not have a link
+     * @param array $options Array of options
+     * @return $this
+     */
+    public function insertAfter($matchingTitle, $title, $link = null, array $options = [])
+    {
+        $found = false;
+        foreach ($this->crumbs as $key => $crumb) {
+            if ($crumb['title'] === $matchingTitle) {
+                $found = true;
+                break;
+            }
+        }
+
+        if ($found) {
+            return $this->insertAt($key + 1, $title, $link, $options);
+        }
+        throw new LogicException(sprintf("No crumb matching '%s' could be found.", $matchingTitle));
+    }
+
+    /**
+     * Returns the crumbs list
+     *
+     * @return array
+     */
+    public function getCrumbs()
+    {
+        return $this->crumbs;
+    }
+
+    /**
+     * Renders the breadcrumbs trail
+     *
+     * @param array $attributes Array of attributes applied to the wrapper element
+     * @param array $separator Array of attributes for the `separator` template
+     * @return string The breadcrumbs trail
+     */
+    public function render(array $attributes = [], array $separator = [])
+    {
+        $crumbs = $this->crumbs;
+        $crumbsCount = count($crumbs);
+        $templater = $this->templater();
+
+        if (!empty($separator)) {
+            $separatorParams = [];
+            if (isset($separator['innerAttrs'])) {
+                $separatorParams['innerAttrs'] = $templater->formatAttributes($separator['innerAttrs']);
+                unset($separator['innerAttrs']);
+            }
+
+            if (isset($separator['separator'])) {
+                $separatorParams['separator'] = $separator['separator'];
+                unset($separator['separator']);
+            }
+
+            $separatorParams['attrs'] = $templater->formatAttributes($separator);
+        }
+
+        $crumbTrail = '';
+        foreach ($crumbs as $key => $crumb) {
+            $link = $this->prepareLink($crumb['link']);
+            $title = $crumb['title'];
+            $options = $crumb['options'];
+
+            $optionsLink = [];
+            if (isset($options['innerAttrs'])) {
+                $optionsLink = $options['innerAttrs'];
+                unset($options['innerAttrs']);
+            }
+
+            $template = 'item';
+            $templateParams = [
+                'attrs' => $templater->formatAttributes($options),
+                'innerAttrs' => $templater->formatAttributes($optionsLink),
+                'title' => $title,
+                'link' => $link,
+                'separator' => '',
+            ];
```

Comment:
`'templateVars' => isset($options['templateVars']) ? $options['templateVars'] : []` should be added to this array so that once can use custom template vars in the string template. Similarly options passed to all call to `formatTemplate()` should account for `templateVars`.